dojo-dungeon/src/dojo/rooms/rooms/riddle/Fragen_Pattern.java

```diff
@@ -103,25 +114,34 @@ private BiConsumer<Task, Set<TaskContent>> showAnswersOnHud() {
       if (answer.matches(EXPECTED_PATTERNS[currentPatternIndex])) {
         OkDialog.showOkDialog("Ihre Antwort ist korrekt!", "Antwort", () -> {});
         correctAnswerCount++;
-        if (correctAnswerCount >= 2) {
+        if (correctAnswerCount == MIN_NUMBER_OF_CORRECT_ANSWERS) {
+          OkDialog.showOkDialog(
+              "Die Tür wird geöffnet, aber Sie können auch noch weiter Fragen beantworten!",
+              "Antwort",
+              () -> {});
           openDoors();
         }
-        currentPatternIndex++;
-        if (currentPatternIndex < EXPECTED_PATTERNS.length) {
-          setNextTask();
-        }
+        setNextTask();
       } else {
-        OkDialog.showOkDialog("Ihre Antwort ist nicht korrekt!", "Ok", () -> {});
+        decreaseHerosHealthAtWrongTry(MAX_NUMBER_OF_WRONG_ANSWERS);
```

Comment:
es sollte noch ein popup geben, wenn die anzahl der fragen überschritten wird: "gehen sie weiter, hier gibt es nichts mehr für sie!"

möglicherweise musst du dafür aber die logik nochmal umdrehen und den current-index einfach immer hochzählen. dann kannst du hier auf `>` prüfen, um festzustellen, dass jemand bereits alles gesehen hat. im nextpattern würde das `if` wegfallen (nur `++`) und beim indexzugriff in der ersten zeile ein modulo hinzukommen.

Reply:
vermutlich sollte das if/else dafür erweitert werden. ich musste nur den kommentar irgendwo dranpappen, und das war die einzige freie stelle. aber hier sollte die änderung nicht direkt rein 🤣